clients/drcachesim/tools/invariant_checker.cpp

```diff
@@ -620,39 +651,35 @@ invariant_checker_t::parallel_shard_memref(void *shard_data, const memref_t &mem
                 memref.instr.addr == shard->last_signal_context_.xfer_int_pc ||
                 // DR hands us a different address for sysenter than the
                 // resumption point.
-                shard->last_signal_context_.pre_signal_instr.instr.type ==
+                shard->last_signal_context_.pre_signal_instr.memref.instr.type ==
                     TRACE_TYPE_INSTR_SYSENTER;
             bool pre_signal_flow_continuity =
                 // Skip pre-signal instr check if there was no such instr. May
                 // happen for nested signals without any intervening instr, and
                 // if the signal arrived before the first instr in the trace.
-                shard->last_signal_context_.pre_signal_instr.instr.addr == 0 ||
+                shard->last_signal_context_.pre_signal_instr.memref.instr.addr == 0 ||
                 // Skip pre_signal_instr_ check for signals that caused an rseq
                 // abort. In this case, control is transferred directly to the abort
                 // handler, verified using last_signal_context_.xfer_int_pc above.
                 shard->last_signal_context_.xfer_aborted_rseq ||
                 // Pre-signal instr continued after signal.
                 memref.instr.addr ==
-                    shard->last_signal_context_.pre_signal_instr.instr.addr ||
+                    shard->last_signal_context_.pre_signal_instr.memref.instr.addr ||
                 // Asynch will go to the subsequent instr.
                 memref.instr.addr ==
-                    shard->last_signal_context_.pre_signal_instr.instr.addr +
-                        shard->last_signal_context_.pre_signal_instr.instr.size ||
-                // Too hard to figure out branch targets.  We have the
-                // last_signal_context_.xfer_int_pc though.
-                // TODO i#5912: since we have the branch decoding now, we can handle
-                // this case.
```

Comment:
From offline discussion: we want to remove this entire `pre_signal_flow_continuity` bool since this A to B check is already being done (and more completely, by checking branches) elsewhere.